src/main/scala/com/ubirch/niomon/healthcheck/HealthCheckServer.scala

```diff
@@ -159,8 +161,15 @@ object Checks {
 
     val processedMetrics = relevantMetrics(metrics)
     // TODO: ask for other failure conditions
-    val success = !connectionCountMustBeNonZero ||
-      processedMetrics.getOrElse("connection-count", 0.0).asInstanceOf[Double] != 0.0
+    val success = (!connectionCountMustBeNonZero ||
+      processedMetrics.getOrElse("connection-count", 0.0).asInstanceOf[Double] != 0.0) &&
+      {
+        val lastHeartbeat = processedMetrics.getOrElse("last-heartbeat-seconds-ago", 0.0).asInstanceOf[Double]
+
+        // the second part of that check is needed because kafka uses unreasonable defaults
+        // (the value is roughly seconds since 1970, as of 2019-10-28)
+        lastHeartbeat < 60.0 || lastHeartbeat > 49 * 365 * 24 * 60 * 60
```

Comment:
The second part of this `||` happens if you try to ask for health before the first heartbeat. This happens for example, in one of the tests for `niomon-foundation`.